src/disk_io_thread.cpp

```diff
@@ -2084,22 +2086,31 @@ namespace libtorrent
 		DLOG("kick_hasher: %d - %d (piece: %d offset: %d)\n"
 			, cursor, end, int(pe->piece), ph->offset);
 
+		// save a local copy of offset to avoid concurrent access
+		int offset = ph->offset;
+#if TORRENT_USE_ASSERTS
+		int old_offset = offset;
```

Comment:
oh, right. my first pass reading this, I assumed ``offset`` was immutable. then I realized it wasn't, but I was still thinking that the assert I proposed would work.. oh well